### PR TITLE
Use native chunker library when available

### PR DESCRIPTION
**Pull request by jerickmsft**
React to some changes in the underlying storage stack.

### CODE REVIEW

File: src/Agent.Plugins/Agent.Plugins.csproj [hunk 1]
```diff
@@ -19,7 +19,15 @@
     <PackageReference Include="Newtonsoft.Json" Version="11.0.2" />
     <PackageReference Include="System.Security.Cryptography.ProtectedData" Version="4.4.0" />
     <PackageReference Include="System.Text.Encoding.CodePages" Version="4.4.0" /> -->
-    <PackageReference Include="vss-api-netcore" Version="0.5.91-private" />
+    <PackageReference Include="ContentStoreInterfaces" Version="0.20181220.6-Dogfood" />
```
Comment (TingluoHuang) [inline src/Agent.Plugins/Agent.Plugins.csproj]: why we need to do this? shouldn't the `vss-api-netcore` has dependency on `ContentStoreInterfaces`?, also why we are using `x-Dogfood` version? does it pass the OSS assets scan?

File: src/Agent.Plugins/Agent.Plugins.csproj [hunk 2]
```diff
@@ -19,7 +19,15 @@
...
+    <PackageReference Include="vss-api-netcore" Version="0.5.94-private" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <None Include="$(NuGetPackageRoot)\DeduplicationSigned\1.0.14\build\net45\**">
```
Comment (TingluoHuang) [inline src/Agent.Plugins/Agent.Plugins.csproj]: why we have to do this manually in csproj? what kind of dependecies are they?

File: src/Test/L0/DotnetsdkDownloadScriptL0.cs [hunk 3]
```diff
@@ -12,7 +12,7 @@ public sealed class DotnetsdkDownloadScriptL0
         [Trait("Category", "Agent")]
         public async Task EnsureDotnetsdkBashDownloadScriptUpToDate()
         {
-            string shDownloadUrl = "https://raw.githubusercontent.com/dotnet/cli/master/scripts/obtain/dotnet-install.sh";
```
Comment (TingluoHuang) [inline src/Test/L0/DotnetsdkDownloadScriptL0.cs]: i am not interesting on change this, the original string can tells you the actual script is on a github repo, and when you are hitting a problem you can open an issue on their repository. :)

Comment (jerickmsft) [reply src/Test/L0/DotnetsdkDownloadScriptL0.cs]: I'm going to bring that one separately.

File: src/Agent.Plugins/Agent.Plugins.csproj (hunk 1 above)
Comment (jerickmsft) [reply src/Agent.Plugins/Agent.Plugins.csproj]: I think we need an explicit PackageReference here for Agent.Plugins to be able to directly reference the types in ContentStoreInterfaces.  This was my intention.

ContentStoreInterfaces is built by the BuildXL team.  Previously we had forked their code as they were not yet build netstandard, but now they are so we deleted the fork to use their build directly.

I'll double check with them on the naming of the package and OSS asset scan. Good call!

File: src/Test/L0/DotnetsdkDownloadScriptL0.cs (hunk 3 above)
Comment (jerickmsft) [reply src/Test/L0/DotnetsdkDownloadScriptL0.cs]: Brought it to #2031 with a better description.

**Comment by TingluoHuang** [comment]
@jerickmsft your new package is causing nuget restore warning in build.
```
2018-12-28T23:40:47.9851764Z D:\a\1\s\src\Microsoft.VisualStudio.Services.Agent\Microsoft.VisualStudio.Services.Agent.csproj : warning NU1603: Microsoft.NETCore.Runtime 1.0.2-rc2-24008 depends on Microsoft.NETCore.Runtime.CoreCLR (>= 1.0.2-rc2-24008) but Microsoft.NETCore.Runtime.CoreCLR 1.0.2-rc2-24008 was not found. An approximate best match of Microsoft.NETCore.Runtime.CoreCLR 1.0.2-rc2-24011 was resolved. [D:\a\1\s\src\dir.proj]
2018-12-28T23:40:47.9865853Z D:\a\1\s\src\Microsoft.VisualStudio.Services.Agent\Microsoft.VisualStudio.Services.Agent.csproj : warning NU1603: Microsoft.NETCore.Runtime 1.0.2-rc2-24008 depends on Microsoft.NETCore.Runtime.Native (>= 1.0.2-rc2-24008) but Microsoft.NETCore.Runtime.Native 1.0.2-rc2-24008 was not found. An approximate best match of Microsoft.NETCore.Runtime.Native 1.0.2-rc2-24011 was resolved. [D:\a\1\s\src\dir.proj]
2018-12-28T23:40:47.9867614Z D:\a\1\s\src\Microsoft.VisualStudio.Services.Agent\Microsoft.VisualStudio.Services.Agent.csproj : warning NU1603: vss-api-netcore 0.5.94-private depends on RuntimeContracts (>= 0.1.5) but RuntimeContracts 0.1.5 was not found. An approximate best match of RuntimeContracts 0.1.6 was resolved. [D:\a\1\s\src\dir.proj]
2018-12-28T23:40:47.9868223Z D:\a\1\s\src\Microsoft.VisualStudio.Services.Agent\Microsoft.VisualStudio.Services.Agent.csproj : warning NU1603: NETStandard.Library 1.0.0-rc2-23910 depends on System.IO.Compression (>= 4.1.0-rc2-24008) but System.IO.Compression 4.1.0-rc2-24008 was not found. An approximate best match of System.IO.Compression 4.1.0-rc2-24011 was resolved. [D:\a\1\s\src\dir.proj]
2018-12-28T23:40:47.9868717Z D:\a\1\s\src\Microsoft.VisualStudio.Services.Agent\Microsoft.VisualStudio.Services.Agent.csproj : warning NU1603: NETStandard.Library 1.0.0-rc2-23910 depends on System.IO.Compression.ZipFile (>= 4.0.1-rc2-24008) but System.IO.Compression.ZipFile 4.0.1-rc2-24008 was not found. An approximate best match of System.IO.Compression.ZipFile 4.0.1-rc2-24011 was resolved. [D:\a\1\s\src\dir.proj]
2018-12-28T23:40:47.9869214Z D:\a\1\s\src\Microsoft.VisualStudio.Services.Agent\Microsoft.VisualStudio.Services.Agent.csproj : warning NU1603: NETStandard.Library 1.0.0-rc2-23910 depends on System.Net.Sockets (>= 4.1.0-rc2-24008) but System.Net.Sockets 4.1.0-rc2-24008 was not found. An approximate best match of System.Net.Sockets 4.1.0-rc2-24011 was resolved. [D:\a\1\s\src\dir.proj]
2018-12-28T23:40:47.9869672Z D:\a\1\s\src\Microsoft.VisualStudio.Services.Agent\Microsoft.VisualStudio.Services.Agent.csproj : warning NU1603: NETStandard.Library 1.0.0-rc2-23910 depends on System.Threading.Timer (>= 4.0.1-rc2-24008) but System.Threading.Timer 4.0.1-rc2-24008 was not found. An approximate best match of System.Threading.Timer 4.0.1-rc2-24011 was resolved. [D:\a\1\s\src\dir.proj]
```

**Comment by jerickmsft** [comment]
@TingluoHuang I think those warnings are cause just from the new package from HEAD - not specific to any changes I have made.

**Comment by TingluoHuang** [comment]
@jerickmsft the changes in this PR includes consuming the new package from VSO master.
Base on the warning, looks like the agent got downgrade to netcore 1.0-rc, i think we need to fix the package before we can consume it in the agent.

**Comment by jerickmsft** [comment]
From the `project.assets.json` it looks like Minimatch is requesting 1.0-rc
```
      "Minimatch/2.0.0": {
        "type": "package",
        "dependencies": {
          "NETStandard.Library": "1.0.0-rc2-23910"
        },
        "compile": {
          "lib/netstandard1.0/Minimatch.dll": {}
        },
        "runtime": {
          "lib/netstandard1.0/Minimatch.dll": {}
        }
      },
```

I've added a commit to bring us to NETStandard 2.0.3.

**Comment by TingluoHuang** [comment]
Why the agent code needs to do that? the agent consume `vss-api-netcore`, ideally i would not expect any change the csproj (other than bump version) when consume newer version package. :)

**Comment by jerickmsft** [comment]
Honestly, I don't know.  That minimatch dependency has been there a long time, so I don't know why it's complaining now.

File: src/Agent.Plugins/Agent.Plugins.csproj (hunk 1 above)
Comment (jerickmsft) [reply src/Agent.Plugins/Agent.Plugins.csproj]: [chatted with Ting]
Does this work if we remove explicit references to RuntimeContracts and ContentStoreInterfaces? (and have them work implicitly via vss-api-netcore)

Comment (jerickmsft) [reply src/Agent.Plugins/Agent.Plugins.csproj]: @TingluoHuang  let me know what you think

**Comment by TingluoHuang** [comment]
@jerickmsft will https://github.com/Microsoft/azure-pipelines-agent/pull/2050 work for you?

**Comment by jerickmsft** [comment]
I think we need to move from minimatch to https://www.nuget.org/packages/Glob/. The former seems dead and doesn't target any official net standard releases.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by jerickmsft** [comment]
Merging what Ting said in the other thread:

"in this case, minimatch requires netcore >=1.0.2-rc2-24008, however 1.0.2-rc2-24008 get delete from dotnet nuget feed. the error tells us it will try find a netcore that is >= 1.0.2-rc2-24011. so i think it's safe to ignore the warning in this case."

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by jerickmsft** [comment]
Thanks, @TingluoHuang!

**Pull request closed without merging**